sdks/java/core/src/main/java/org/apache/beam/sdk/util/ExplicitShardedFile.java

```diff
@@ -49,18 +49,36 @@
           .withMaxRetries(MAX_READ_RETRIES);
 
   private final List<Metadata> files;
+  private final String filePattern;
 
   /** Constructs an {@link ExplicitShardedFile} for the given files. */
   public ExplicitShardedFile(Collection<String> files) throws IOException {
     this.files = new ArrayList<>();
+    this.filePattern = null;
     for (String file : files) {
       this.files.add(FileSystems.matchSingleFileSpec(file));
     }
   }
 
+  /**
+   * Constructs an {@link ExplicitShardedFile} for the given file pattern.
+   *
+   * <p>For internal use only.
+   */
+  public ExplicitShardedFile(String filePattern) {
```

Comment:
Adding this constructor is specifically against why this class exists. The intent is to not require a filesystem match.